### PR TITLE
DOCS: use nbsphinx-link and include myst beta

### PR DESCRIPTION
**Pull request by knaaptime**
this resolves the issues the the docs (locally anyway):

- fixes the bib not rendering
- uses nbsphinx-link (correctly this time, because the makefile no longer interferes)
- includes a beta of myst and autodoc2. This isn't exposed yet as a link but will be rendered at pysal.org/tobler/apimyst for comparison when the docs are built
	- I don't love the style, but I hate RST and maybe numpydoc style will be supported in the future
	- this required making the private/public functions more explicit using `__all__` throughout

### CODE REVIEW

**Comment by jGaboardi** [comment]
@knaaptime And the [macOS13](https://github.com/pysal/tobler/actions/runs/11451230406/job/31860090514) failure is ongoing for awhile now so we don't have worry about here, right?

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -2,47 +2,41 @@
...
+     name: Build & Push Docs
      runs-on: ${{ matrix.os }}
-     continue-on-error: ${{ matrix.experimental }}
-     timeout-minutes: 20
+     timeout-minutes: 90
      strategy:
        matrix:
          os: ['ubuntu-latest']
-         environment-file: [ci/311.yml]
+         environment-file: [ci/311.yaml]
```
Comment (jGaboardi) [inline .github/workflows/build_docs.yml:15]: Should we try for 3.12 here?

Comment (knaaptime) [reply .github/workflows/build_docs.yml:15]: sure, we can bump one version. There's a bunch of extra crap for the docs in that env

**Comment by knaaptime** [comment]
yeah the failure at the moment is actually a codecov timeout or something

**Comment by jGaboardi** [comment]
Is there a preview somewhere that is rendered?

**Comment by knaaptime** [comment]
if you build locally and launch a webserver in side the _html dir, its the `apimyst.html` page 
Screenshot 2024-10-30 at 6 13 37 PM
Screenshot 2024-10-30 at 6 13 14 PM

File: docs/apimyst.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# API 
+
+This is a beta version of the documentation written in [MyST](https://myst-parser.readthedocs.io/en/latest/index.html) with [autodoc2](https://sphinx-autodoc2.readthedocs.io/en/latest/quickstart.html)
+
+## Area Weighted Interpolation
+
+Area weighted approaches use the area of overlap between the source and target geometries to weight the variables being assigned to the target
```
Comment (jGaboardi) [inline docs/apimyst.md:7]:
```suggestion
Area weighted approaches use the area of overlap between the source and target geometries to weight the variables being assigned to the target.
```

Need proper punctuation or no? Maybe not?

File: docs/apimyst.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+```{autodoc2-summary}  tobler.area_weighted.area_join.area_join
+```
+
+## Dasymetric Interpolation
+
+Dasymetric approaches use auxiliary data in addition to use the area of overlap
+between the source and target geometries to weight the variables being assigned
+to the target
```
Comment (jGaboardi) [inline docs/apimyst.md:19]:
```suggestion
to the target.
```

Need proper punctuation or no? Maybe not?

File: docs/apimyst.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+```{autodoc2-summary}  tobler.dasymetric.masked_area_interpolate.masked_area_interpolate
+```
+
+## Pycnophylactic Interpolation
+
+Pycnophylactic interpolation is based on
+[Tobler's technique](https://www.tandfonline.com/doi/abs/10.1080/01621459.1979.10481647)
+for generating smooth, volume-preserving contour maps
```
Comment (jGaboardi) [inline docs/apimyst.md:31]:
```suggestion
for generating smooth, volume-preserving contour maps.
```

**Comment by jGaboardi** [comment]
[Windows failure](https://github.com/pysal/tobler/actions/runs/11621264714/job/32365029610?pr=219) is a bit cryptic. Any idea what's going on there?

**Comment by martinfleis** [comment]
> Any idea what's going on there?

ongoing micromamba action issue. ignore it - https://github.com/mamba-org/setup-micromamba/issues/227

**Review by jGaboardi — APPROVED** [review]

**Comment by knaaptime** [comment]
you can see the live versions created by autodoc2 at https://pysal.org/tobler/apimyst.html and  https://pysal.org/tobler/apidocs

**Review by sjsrey — APPROVED** [review]

**Pull request merged**